lib/brakeman/options.rb

```diff
@@ -325,7 +329,7 @@ def create_option_parser options
           options[:min_confidence] =  3 - level.to_i
         end
 
-        opts.on "--compare FILE", "Compare the results of a previous Brakeman scan (only JSON is supported)" do |file|
+        opts.on "--compare FILE", "Compare the results of a previous Brakeman scan (only JSON is supported). Deactivates --ensure-ignore-notes" do |file|
```

Comment:
Instead of documenting here (which makes the help line pretty long), what about detecting that two conflicting options were given?

That could be added in `Commandline#parse_options`.

Reply:
I structured this so that we fail in favor of the user's likely intention, and simply de-activate `--ensure-ignore-notes` if it is used in conjunction with `--compare`, outputting the following warning (and then continuing to process the comparison):
```
Warning: --ensure-ignore-notes may not be used at the same time as --compare. Deactivating --ensure-ignore-notes. Please see `brakeman --help` for valid options
```

Let me know if you think it would be better to exit with an error in this case!